app/routes/events/components/EventAttendeeStatistics.tsx

```diff
@@ -253,13 +259,36 @@ const Analytics = ({ eventId }: { eventId: ID }) => {
 
   const dispatch = useAppDispatch();
 
+  const memoizedData = useMemo(() => {
+    return async () => {
+      if (eventId) {
+        const response = await dispatch(fetchAnalytics(eventId));
+        let data = response.payload;
+
+        if (viewStartTime) {
+          data = data.filter(
+            (item) => new Date(item.date) >= new Date(viewStartTime as string)
+          );
+        }
+
+        if (viewEndTime) {
+          data = data.filter(
+            (item) => new Date(item.date) <= new Date(viewEndTime as string)
+          );
+        }
+        return data;
+      }
+    };
+  }, [eventId, viewStartTime, viewEndTime, dispatch]);
+
   useEffect(() => {
-    eventId &&
-      dispatch(fetchAnalytics(eventId)).then((res) => {
-        setData(res.payload);
-        setMetrics(calculateMetrics(res.payload));
+    if (eventId) {
+      memoizedData().then((data) => {
+        setData(data);
+        setMetrics(calculateMetrics(data));
       });
-  }, [eventId, dispatch]);
+    }
+  }, [eventId, memoizedData]);
```

Comment:
Because you are memoizing the data based on `eventId`, you don't need this `useEffect` to also depend on `eventId` :smile: 
```suggestion
  }, [memoizedData, viewStartTime, viewEndTime]);
```

Reply:
Seems like I have to for some reason... CI fails when I remove `eventId` with this message:
```
React Hook useEffect has a missing dependency: 'eventId'. Either include it or remove the dependency array
```